x-pack/legacy/plugins/infra/public/containers/logs/log_analysis/log_analysis_module_status.tsx

```diff
@@ -65,42 +64,60 @@ type StatusReducerAction =
   | { type: 'requestedJobDefinitionUpdate' }
   | { type: 'viewedResults' };
 
-const createInitialState = (sourceConfiguration: JobSourceConfiguration): StatusReducerState => ({
+const createInitialState = <JobType extends string>({
+  jobTypes,
+  sourceConfiguration,
+}: {
+  jobTypes: JobType[];
+  sourceConfiguration: JobSourceConfiguration;
+}): StatusReducerState<JobType> => ({
   jobDefinitions: [],
-  jobStatus: {
-    'log-entry-rate': 'unknown',
-  },
+  jobStatus: jobTypes.reduce(
+    (accumulatedJobStatus, jobType) => ({
+      ...accumulatedJobStatus,
+      [jobType]: 'unknown',
+    }),
+    {} as Record<JobType, JobStatus>
```

Comment:
Small nitpick: Could we use a generic type argument over the casting? (`jobTypes.reduce<Record<JobType, JobStatus>>`)?

Reply:
That's what I usually prefer too, but in this case it doesn't work because the empty default object is not assignable to that record type:

```
Argument of type '{}' is not assignable to parameter of type 'Record<JobType, JobStatus>'.
```